AEPTarget/Sources/Target.swift

```diff
@@ -536,6 +619,66 @@ public class Target: NSObject, Extension {
         }
     }
 
+    private func processTargetRawRequestResponse(event: Event, connection: HttpConnection) {
+        var error: String? = (connection.responseMessage != TargetConstants.NetworkConnection.CONNECTION_RESPONSE_MESSAGE_NO_ERROR) ? connection.responseMessage : nil
+        var responseData: [[String: Any]]?
+
+        defer {
+            dispatchExecuteRawResponse(event: event, error: error, data: responseData)
+        }
+
+        guard
+            let data = connection.data,
+            let responseDictAnyCodable = try? JSONDecoder().decode([String: AnyCodable].self, from: data),
+            let responseDict = AnyCodable.toAnyDictionary(dictionary: responseDictAnyCodable)
+        else {
+            Log.debug(label: Target.LOG_TAG, "Target response parser initialization failed")
+            error = TargetError.ERROR_RESPONSE_PARSING_FAILED
+            return
+        }
+
+        let response = TargetDeliveryResponse(responseJson: responseDict)
+
+        if connection.responseCode != 200 {
+            if let responseError = response.errorMessage {
+                if responseError.contains(TargetError.ERROR_NOTIFICATION_TAG) {
+                    targetState.clearNotifications()
+                }
+                error = responseError
+            } else {
+                error = connection.error?.localizedDescription ?? error
+            }
+            Log.debug(label: Target.LOG_TAG, "Target raw execute request failed with response code: \(String(describing: connection.responseCode)) and error: \(String(describing: error)))")
+            return
+        }
+
+        // Clear the notifications when the response is 200 OK
+        targetState.clearNotifications()
+
+        if let tntId = response.tntId { targetState.updateTntId(tntId) }
+        if let edgeHost = response.edgeHost { targetState.updateEdgeHost(edgeHost) }
+        createSharedState(data: targetState.generateSharedState(), event: nil)
+
+        responseData = response.executeMboxes
+    }
+
+    private func dispatchExecuteRawResponse(event: Event, error: String?, data: [[String: Any]]?) {
+        Log.trace(label: Target.LOG_TAG, "Dispatching raw response for Target execute request.")
+
+        var eventData = [String: Any]()
+        if let data = data {
+            eventData[TargetConstants.EventDataKeys.EXECUTE_MBOXES] = data as Any
```

Comment:
nit: might be best to add the cast to the condition rather than doing it during assignment

Reply:
The downcast to Any is implicit here so I can just omit it.

Reply:
gotcha, i assumed you were doing it to get rid of a compiler warning